static/backbone/js/api/views/maps/overlays/layer.js

```diff
@@ -0,0 +1,152 @@
+                    overlay.redraw();
+                });
+            },
+            render: function () {
+                //console.log("rendering ", this.model.get("name"));
+                var rule;
+                for (rule in this.overlayMap) {
+                    this.renderSymbol(rule);
+                }
+            },
+            parseLayerItem: function () {
+                var that = this;
+                _.each(this.model.getSymbols(), function (symbol) {
+                    //clear out old overlays and models
+                    that.clear(symbol);
+                    _.each(_.values(that.dataManager.collections), function (collection) {
+                        that.addMatchingModels(symbol, collection);
+                    });
+                });
+            },
+            clear: function (symbol) {
+                symbol.models = [];
+                this.hideSymbol(symbol.rule);
+                //once hidden, remove all symbolized map overlay objects for g.c.
+                this.overlayMap[symbol.rule] = [];
+            },
+            addMatchingModels: function (symbol, collection) {
+                var match = false,
+                    that = this;
+                collection.each(function (model) {
+                    match = symbol.checkModel(model);
+                    if (match) {
+                        symbol.addModel(model);
+                        that.addOverlay(symbol, model);
+                    }
+                });
+            },
+            addOverlay: function (symbol, model) {
+                var configKey,
+                    opts;
+                if (model.get('geometry') != null) {
+                    configKey = model.getKey().split("_")[0];
+                    opts = {
+                        app: this.app,
+                        model: model,
+                        symbol: symbol,
+                        map: this.map,
+                        infoBubbleTemplates: {
+                            InfoBubbleTemplate: _.template(Config[configKey].InfoBubbleTemplate),
+                            TipTemplate: _.template(Config[configKey].TipTemplate)
+                        }
+                    };
+                    if (this.overlayMap[symbol.rule] == null) {
+                        this.overlayMap[symbol.rule] = [];
+                    }
+                    this.overlayMap[symbol.rule].push(new Symbolized(opts));
+                }
+            },
+
+            hideSymbol: function (rule) {
+                var i;
+                if (!this.overlayMap[rule]) {
+                    return;
+                }
+                for (i = 0; i < this.overlayMap[rule].length; i++) {
+                    this.overlayMap[rule][i].hide();
+                }
+            },
+
+
+            /** Zooms to the extent of the collection */
+            zoomToExtent: function () {
+                console.log("zoom to extent");
```

Comment:
I really tried to come up with a 'more like console.bog!' style pun on log about how it should be taken out but couldn't so just pretend I wrote something witty and take out the console.log :^] (also a couple other scattered logs currently commented out)